src/DGtal/topology/doc/moduleCubicalComplex.dox

```diff
@@ -0,0 +1,328 @@
+
+/* 
+ * Useful to avoid writing DGtal:: in front of every class.
+ */
+namespace DGtal {
+
+/**
+   
+@page moduleCubicalComplex  Cubical Complex
+
+@writers Jacques-Olivier Lachaud
+
+@since 0.9.1
+
+Part of the \ref packageTopology.
+  
+This part of the manual describes how to represent and process
+arbitrary cubical complexes. 
+
+@note Collapse operation is a backport from <a
+href="https://gforge.liris.cnrs.fr/projects/imagene">ImaGene</a>. \cite
+ImaGene 
+
+[TOC]
+
+
+The following programs are related to this documentation:
+cubical-complex-collapse.cpp, cubical-complex-illustrations.cpp,
+testCubicalComplex.cpp,
+
+@section dgtal_ccomplex_sec1  Introduction to cubical complexes
+
+We define a \b cubical \b complex \a C as a collection of cells living
+in some Khalimsky space. Two cells of \a C are \b incident if and only
+if they are incident in the Khalimsky space. 
+
+@note In opposition to the usual definition of simplical complex, we
+do not require that all faces of a cell of \a K belong also to \a K.
+
+To create a cubical complex, we need to specify in which Khalimsky
+space it lives and also, optionally, the type of container used for
+storing cells. By default it is \c std::map but \c boost::unordered_map
+or \c std::unordered_map is also possible.
+
+\code
+using namespace DGtal;
+typedef KhalimskySpaceND<2,int>           KSpace; // ZxZ
+typedef std::map<Cell, CubicalCellData>   Map;    // each cell must store at least a CubicalCellData
```

Comment:
I've done it. I've also fixed a problem in the rebind structure in KhalimskySpaceND. The rebind syntax for a default template parameter is a pain in the ass !